vale/main.py

```diff
@@ -109,19 +115,27 @@ def download_vale_if_missing() -> str:
 
         url = urlopen(url)
 
-        with tempfile.NamedTemporaryFile(mode="w+b") as tp:
+        # delete=False is required to avoid permissions errors on windows
+        with tempfile.NamedTemporaryFile(mode="w+b", delete=False) as tp:
 
             tp.write(url.read())
 
             print(f"* {url} downloaded to {tp.name}")
 
             with tempfile.TemporaryDirectory() as td:
 
-                vale_tmp_path = extract_vale(tp.name, extension, td)
+                archive_bin_name = "vale.exe" if operating_system == "Windows" else "vale"
+                vale_tmp_path = extract_vale(tp.name, extension, td, archive_bin_name)
 
                 print(f"* Copying {vale_tmp_path} to {vale_bin_path}")
                 shutil.copy(f"{vale_tmp_path}", f"{vale_bin_path}")
 
+        # clean up the temp file if it still exists
+        try:
+            os.unlink(tp.name)
+        except Exception:
+            pass
```

Comment:
I think your solution is good, especially taking into account that the current version fails with Windows anyway! 

Perhaps, in the future, I'll create a `TemporaryDirectory` that remains open as long as `vale` is not correctly downloaded and in place. Then I can create files in that folder without needing to worry about whether they are open once or twice, or whether they have been removed or not.

Reply:
Yes, that sounds quite sensible to me! It's a real shame that NamedTemporaryFile behaves the way it does; the current workaround basically defeats the elegance of the context manager.